src/components/Settings.jsx

```diff
@@ -86,6 +92,44 @@ export default function Settings({ currentWallet }) {
           />
           Use {settings.useAdvancedWalletMode ? 'magic' : 'advanced'} wallet mode
         </rb.Button>
+
+        <br />
+
+        <rb.Button
+          variant="outline-dark"
+          className="border-0 mb-2 d-inline-flex align-items-center"
+          onClick={async (e) => {
+            e.preventDefault()
+            setRevealSensitiveInfo(false)
+            const { name: walletName, token } = currentWallet
+            const res = await Api.getSeed({ walletName, token })
+            if (res.ok) {
+              const { seedphrase } = await res.json()
+              setSeed(seedphrase)
+              setShowSeed(!showSeed)
+            } else {
+              console.warn('Could not retrieve seedphrase')
+            }
+          }}
+        >
+          <Sprite symbol="mnemonic" width="24" height="24" className="me-2" />
+          {showSeed ? 'Hide' : 'Show'} seed phrase backup
+        </rb.Button>
+        {showSeed && (
+          <div style={{ marginLeft: '1rem' }}>
+            <div className="mb-4">
+              <Seedphrase seedphrase={seed} isBlurred={!revealSensitiveInfo} />
+            </div>
+            <div className="mb-2">
+              <ToggleSwitch
+                label="Reveal sensitive information"
+                onToggle={(isToggled) => {
+                  setRevealSensitiveInfo(isToggled)
+                }}
+              />
+            </div>
+          </div>
```

Comment:
Just a thought: Maybe a pop up that shows the seed with some additional warning and confirmation step before would be a better fit UI-wise here. But the current one is fine as well. I would add some warning text to the toggle, though.

Reply:
What kind of warning text would you like for the toggle? I was thinking 'reveal sensitive information' lets the user know that it is important, should we add something on top of this?

Reply:
> Just a thought: Maybe a pop up that shows the seed with some additional warning and confirmation step before would be a better fit UI-wise here. 

Would [`<rb.Modal>`](https://react-bootstrap.github.io/components/modal/#vertically-centered) would be a good fit for that?

Reply:
We can do that in an improvement issue as well though.

Reply:
Ok sounds good